### DISCUSSION TITLE
npm init when npm install is being executed.

### DISCUSSION THREAD

**Discussion by pshrimal000** [npm/feedback#171] [CLI] [answered]
If a user types npm install <package name>, it is quite obvious that user wants to use npm in his/her project , so any npm install command should trigger npm init command.

**Comment by ljharb** [comment]
It’s not obvious to me; I use npm install in a temp dir all the time without a package.json.

**Comment by pshrimal000** [comment]
yeah you are right but package.json is important file to make the sharing of code easier.

**Reply by ljharb** [reply]
It very much is! Outputting a warning when there's no package.json might be useful.

**Comment by MylesBorins** [comment, marked as answer]
I don't know that there is something directly actionable here. With the way the `node_modules` algorithm works in Node.js and with how npm work npm will search up until the root directory to find a package.json... which allow you to run a command like `npm install` from anywhere within a project and have it work as expected (e.g. use the closest package.json).

If you run `npm install` in a directory that has no package.json and none of the folders between it and the root directory have a package.json it does pretty decent behavior that differs slightly between npm 6 and npm 7

npm 6 will create a `package-lock.json` and a `node_modules` folder in the directory you ran the command
npm 7 will create a `package-lock.json` a `node_modules` folder, and a `package.json` with a single dependency in the directory you ran the command.

It seems like a potentially actionable request would be to have npm 7 also include other defaults in the package.json, but I think the current behavior is pretty reasonable

**Reply by ljharb** [reply]
I was unaware of npm 7's behavior here. Presumably `--no-save` would avoid creation of the `package.json`?

**Reply by MylesBorins** [reply]
I can confirm that running with `--no-save` with npm 7 results in no package.json or package-lock.json
